src/library_pthread.js

```diff
@@ -1045,22 +1045,22 @@ var LibraryPThread = {
   },
 
   $executeNotifiedProxyingQueue: function(queue) {
-    try {
-      // Only execute the queue if we have a live pthread runtime. We
-      // implement pthread_self to return 0 if there is no live runtime.
-      // TODO: Use `callUserCallback` to correctly handle unwinds, etc. once
-      //       `runtimeExited` is correctly unset on workers.
-      if (_pthread_self()) {
-        _emscripten_proxy_execute_queue(queue);
-      }
-    } finally {
-      // Always decrement the ref count.
-      Atomics.sub(HEAP32, queue >> 2, 1);
+    // Set the notification state to processing.
+    Atomics.store(HEAP32, queue >> 2, 1);
```

Comment:
cDefine could be used for the constants here, if you move the enum declaration into a header (we have support for non-public headers there, in `src/struct_info_internal.json` IIANM).

Reply:
I had to put the header in a weird internal musl directory, but otherwise this seems to work great.